crates/subspace-farmer/src/utils/archival_storage_pieces.rs

```diff
@@ -47,4 +56,10 @@ impl PeerInfoProvider for ArchivalStoragePieces {
             data: Some(data),
         }
     }
+
+    fn subscribe(&self, handler: NotificationHandler) -> Option<HandlerId> {
```

Comment:
Generally convention is to use `fn on_notification` instead of abstract `fn subscribe` and instead of type alias for type alias have a generic `callback` argument here, but for generic you'll probably have to use GATs.

Reply:
I think it is fine without generic for now, but why did you decide to return an option from here?